app/src/main/kotlin/com/simplemobiletools/flashlight/screens/MainScreen.kt

```diff
@@ -0,0 +1,245 @@
+                contentDescription = stringResource(id = R.string.flashlight_short),
+                tint = if (flashlightActive) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.onSurface
+            )
+
+            if (showBrightDisplayButton) {
+                Icon(
+                    modifier = Modifier
+                        .size(dimensionResource(id = R.dimen.smaller_button_size))
+                        .padding(vertical = dimensionResource(id = R.dimen.normal_margin))
+                        .clickable(
+                            indication = null,
+                            interactionSource = remember {
+                                MutableInteractionSource()
+                            },
+                            onClick = onBrightDisplayPress
+                        ),
+                    painter = painterResource(id = R.drawable.ic_bright_display_vector),
+                    contentDescription = stringResource(id = R.string.bright_display),
+                    tint = MaterialTheme.colorScheme.onSurface
+                )
+            }
+
+            if (showSosButton) {
+                Text(
+                    modifier = Modifier
+                        .padding(vertical = dimensionResource(id = R.dimen.normal_margin))
+                        .clickable(
+                            indication = null,
+                            interactionSource = remember {
+                                MutableInteractionSource()
+                            },
+                            onClick = onSosButtonPress
+                        ),
+                    text = "SOS",
+                    fontSize = TextUnit(dimensionResource(id = R.dimen.sos_text_size).value, TextUnitType.Sp),
+                    fontWeight = FontWeight.Bold,
+                    color = if (sosActive) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.onSurface
+                )
+            }
+
+            if (showStroboscopeButton) {
+                Icon(
+                    modifier = Modifier
+                        .size(dimensionResource(id = R.dimen.smaller_button_size))
+                        .padding(vertical = dimensionResource(id = R.dimen.normal_margin))
+                        .clickable(
+                            indication = null,
+                            interactionSource = remember {
+                                MutableInteractionSource()
+                            },
+                            onClick = onStroboscopeButtonPress
+                        ),
+                    painter = painterResource(id = R.drawable.ic_stroboscope_vector),
+                    contentDescription = "",
+                    tint = if (stroboscopeActive) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.onSurface
+                )
+            }
+
+            val sliderModifier = Modifier
```

Comment:
you can remember it to avoid re-allocating it, but you can't since you're referencing the dimensions